modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/networkapi/NetworkApiService.scala

```diff
@@ -0,0 +1,62 @@
+package org.tessellation.rosetta.domain.networkapi
+
+import cats.data.EitherT
+import cats.effect.Async
+
+import org.tessellation.BuildInfo
+import org.tessellation.rosetta.domain.NetworkIdentifier
+import org.tessellation.rosetta.domain.error.{LatestSnapshotNotFound, NetworkApiError}
+import org.tessellation.rosetta.domain.networkapi.model.options._
+import org.tessellation.rosetta.domain.networkapi.model.status._
+import org.tessellation.schema.GlobalSnapshot
+import org.tessellation.schema.epoch.EpochProgress
+import org.tessellation.security.Hashed
+import org.tessellation.security.hex.Hex
+
+trait NetworkApiService[F[_]] {
+  def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]]
+  def options: EitherT[F, NetworkApiError, NetworkApiOptions]
+  def status: EitherT[F, NetworkApiError, NetworkStatusResponse]
+}
+
+object NetworkApiService {
+  private val rosettaVersion = "1.4.14" // TODO get from a config?
+  private val version = Version(rosettaVersion = rosettaVersion, nodeVersion = BuildInfo.version)
+
+  def make[F[_]: Async](getLastSnapshot: () => F[Option[Hashed[GlobalSnapshot]]]): NetworkApiService[F] = new NetworkApiService[F] {
+    def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]] =
+      EitherT.pure(List(NetworkIdentifier.Mainnet, NetworkIdentifier.Testnet))
+
+    def options: EitherT[F, NetworkApiError, NetworkApiOptions] =
+      EitherT.pure(NetworkApiOptions(version, Allow.default))
+
+    def status: EitherT[F, NetworkApiError, NetworkStatusResponse] =
+      EitherT
+        .fromOptionF[F, NetworkApiError, Hashed[GlobalSnapshot]](getLastSnapshot(), LatestSnapshotNotFound)
+        .map { lastSnapshot =>
+          val genesisSnapshot = GlobalSnapshot.mkGenesis(Map.empty, EpochProgress.MinValue) // TODO confirm this is correct
+
+          val genesisBlockId = BlockIdentifier(
+            genesisSnapshot.height, // TODO confirm: rosetta branch used .ordinal
+            Hex(genesisSnapshot.lastSnapshotHash.value) // TODO confirm this is correct
+          )
+
+          NetworkStatusResponse(
+            currentBlockIdentifier = BlockIdentifier(
+              lastSnapshot.height, // TODO confirm: rosetta branch used .ordinal
+              Hex(lastSnapshot.hash.value)
+            ),
+            currentBlockTimestamp = lastSnapshot.ordinal.value.value, // TODO is this correct?
```

Comment:
This needs to be injected as a dependency / parameter instead, i.e. getTimestampForSnapshot(ordinal) -- we're going to have to query block explorer API (for timestamps,) until schema modifications are done to add timestamps. This should be supplied as a function -- for tests it's fine to use ordinal as a placeholder but that should be done thru a trait